src/Http/Event/UpdateLocationRequestHandler.php

```diff
@@ -33,6 +33,12 @@ public function handle(ServerRequestInterface $request): ResponseInterface
         $eventId = $routeParameters->getEventId();
         $locationId = $routeParameters->get('locationId');
 
+        if ((new LocationId($locationId))->isVirtualLocation()) {
+            throw ApiProblem::pathParameterInvalid(
+                'Instead of passing the virtual location, please update the attendance mode.'
```

Comment:
The main reason is that we don't have to expose/handle the virtual location. An API integrator never has to use the virtual location, not with the import/create event endpoint and not with the update attendance mode endpoint.

Reply:
Also the virtual location is something we partly invented to make sure every event has a location.

Reply:
> an extra safety to make sure everything related to attendance mode goes through the attendance mode endpoint

But then should we also not throw an error when trying to set a real location on an online event, instead of automatically setting the attendanceMode to `offline`? Which is also a guess, could be that the integrator intended `mixed`.

Reply:
All errors is fine for me, I will also discuss with @corneelwille

Reply:
The code is updated to throw an exception